packages/charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx

```diff
@@ -213,25 +210,23 @@ export class VerticalStackedBarChartBase extends React.Component<
   };
 
   private _getFormattedLineData = (data: IVerticalStackedChartProps[]): LineObject => {
-    const linesData: LineData[] = [];
+    const linesData: LinePoint[] = [];
     const formattedLineData: LineObject = {};
     data.forEach((item: IVerticalStackedChartProps, index: number) => {
       if (item.lineData) {
-        linesData.push(...item.lineData);
         // injecting corresponding x data point in each of the line data
         // we inject index also , it will be helpful to draw lines when x asix is
         // of string type
-        linesData.forEach((line: LineData) => {
-          if (line.x === undefined) {
-            line.x = item.xAxisPoint;
-          }
-          if (line.index === undefined) {
-            line.index = index;
-          }
+        item.lineData.forEach(line => {
```

Comment:
could you please revert this change. we need to set `x` value and `index` value only when it is undefined, this new code will override it. which we don't want

Reply:
My new code makes a copy of the objects, so it will need to set these each time this function is run.

Reply:
BTW, the previous code was modifying the objects (props) passed into this component, which is generally frowned upon.

Reply:
@shivasai09 can you make sure this does not break our code?

Reply:
sure.

Reply:
i have noticed one issue. and mentioned in below comments